source/solvers/gls_navier_stokes.cc

```diff
@@ -1545,6 +1550,62 @@ GLSNavierStokesSolver<dim>::solve_system_direct(const bool   initial_step,
   newton_update       = completely_distributed_solution;
 }
 
+
+template <int dim>
+void
+GLSNavierStokesSolver<dim>::solve_system_TFQMR(const bool   initial_step,
+                                               const double absolute_residual,
+                                               const double relative_residual,
+                                               const bool   renewed_matrix)
+{
+  auto &system_rhs          = this->get_system_rhs();
+  auto &nonzero_constraints = this->get_nonzero_constraints();
+
+  const AffineConstraints<double> &constraints_used =
+    initial_step ? nonzero_constraints : this->zero_constraints;
+  const double linear_solver_tolerance =
+    std::max(relative_residual * system_rhs.l2_norm(), absolute_residual);
+
+  if (this->nsparam.linear_solver.verbosity != Parameters::Verbosity::quiet)
+    {
+      this->pcout << "  -Tolerance of iterative solver is : "
+                  << linear_solver_tolerance << std::endl;
+    }
+  TrilinosWrappers::MPI::Vector completely_distributed_solution(
+    this->locally_owned_dofs, this->mpi_communicator);
+
+  SolverControl solver_control(this->nsparam.linear_solver.max_iterations,
+                               linear_solver_tolerance,
+                               true,
+                               true);
+
+  TrilinosWrappers::SolverTFQMR::AdditionalData solver_parameters(false);
+
+
+  TrilinosWrappers::SolverTFQMR solver(solver_control, solver_parameters);
```

Comment:
As in line 1544 and in the deal.ii documentation, I think you don't need to initialize solver_parameters since the TrilinosWrappers::SolverTFQMR::AdditionalData argument is declared and = false in the function.